bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java

```diff
@@ -43,6 +48,8 @@
 
     private Configuration configuration = new Configuration();
 
+    private Map<String, String> properties = new HashMap<>();
```

Comment:
Why is the "properties" feature typed as Map<> while the "configuration" feature has its own Configuration type? Though their semantics is completely different, from a design perspective theses features are very similar.
Try to imagine how client will use this API: thing.setProperty(k, v):void vs. thing.getConfiguration().put(k, v): oldValue
Even if I'd prefer the Map interface, the asymmetric solution is not nice.